### PR TITLE
[RHOAIENG-2404] Replace deprecated PF components

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
Towards: [RHOAIENG-2404](https://issues.redhat.com/browse/RHOAIENG-2404)

## Description
Remove use of Patternfly deprecated componets

## How Has This Been Tested?
Tested manually

## Test Impact
The following components are effected:

- Resources -> Sort type select and sort order select
- Model Serving -> click on a model name: `Time range` and `Refresh interval` select drop downs.
- Data Science Pipelines -> Project select
- Data Science Pipelines -> Filter type select
- Data Science Pipelines -> click a pipeline: Actions select.
- Data Science Pipelines -> Actions -> View schedules -> click a schedule -> Actions select
- Data Science Pipelines -> expand a pipeline -> kebab from a pipeline version -> view runs -> click a run -> Action select
- Data Science Pipelines -> expand a pipeline -> kebab from a pipeline version -> view runs -> click a run -> select a node -> side panel logs tab -> download button
- Experiments -> Experiments and runs -> click an experiment -> select a run -> Compare runs -> change filter to Experiment -> Search input
- Data Science Projects -> select a project -> permissions tab -> Add user -> username selector
- Data Science Projects -> select a project -> permissions tab -> Add group -> group name selector
- Data Science Projects -> select a project -> workbench tab -> create a workbench -> Environment variables -> Add variable -> Select environment variable type
- Data Science Projects -> select a project -> workbench tab -> create a workbench -> Environment variables -> Add variable -> Select environment variable type -> secret -> either -> select one
- Data Science Projects -> select a project -> workbench tab -> create a workbench -> Use a data connection -> use existing data connection -> Select data connection
- Data Science Projects -> select a project -> workbench tab -> create a workbench -> Cluster storage -> use existing persistent storage -> Select a persistent storage
- Science Projects -> select a project -> Cluster storage tab -> Add cluster storage -> connected workbench select
- Data Science Projects -> Launch Jupyter -> select container size

## Request review criteria:
Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

### CODE REVIEW

**Review by christianvogt — COMMENTED** [review]
Still in review, but committing the existing comments as one is a bug in behavior.

File: frontend/src/__tests__/cypress/cypress/tests/mocked/projects/workbench.cy.ts [hunk 1]
```diff
@@ -398,11 +398,15 @@ describe('Workbench page', () => {
...
+    cy.interceptK8sList(
+      PVCModel,
+      mockK8sResourceList([mockPVCK8sResource({ name: 'test-notebook' })]),
+    );
     editSpawnerPage.visit('test-notebook');
     editSpawnerPage.findNameInput().should('have.value', 'Test Notebook');
     editSpawnerPage.shouldHaveNotebookImageSelectInput('Test Image');
     editSpawnerPage.shouldHaveContainerSizeInput('Small');
-    editSpawnerPage.shouldHavePersistentStorage('test-notebook');
+    editSpawnerPage.shouldHavePersistentStorage('Test Storage');
```
[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply frontend/src/__tests__/cypress/cypress/tests/mocked/projects/workbench.cy.ts:409]: There are some inconsistencies in the test data. The way the code was before put a value in that was not necessarily in the selections that are possible for the user to select.

File: frontend/src/__tests__/cypress/cypress/pages/distributedWorkloads.ts [hunk 2]
```diff
@@ -43,7 +43,8 @@ class GlobalDistributedWorkloads {
   }
 
   selectProjectByName(name: string) {
-    this.findProjectSelect().findDropdownItem(name).click();
+    this.findProjectSelect().click();
+    cy.findByRole('menuitem', { name }).click();
```
[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply frontend/src/__tests__/cypress/cypress/pages/distributedWorkloads.ts:47]: correct. this is not necessary.

File: frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/ServingRuntimeSizeSection.tsx [hunk 3]
```diff
@@ -88,23 +86,19 @@ const ServingRuntimeSizeSection: React.FC<ServingRuntimeSizeSectionProps> = ({
       >
         <Stack hasGutter>
           <StackItem>
-            <Select
-              id="model-server-size-selection"
-              isOpen={sizeDropdownOpen}
-              placeholderText="Select a model server size"
-              onToggle={(e, open) => setSizeDropdownOpen(open)}
+            <SimpleSelect
```
Comment (christianvogt) [inline frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/ServingRuntimeSizeSection.tsx:89]: When i first tab to this select and press enter, i cannot use the arrow keys to change options.
But after clicking with the mouse an option, the same steps above work to use arrow keys to change options.

File: frontend/src/pages/projects/screens/spawner/dataConnection/ExistingDataConnectionField.tsx [hunk 4]
```diff
@@ -15,11 +16,27 @@ const ExistingDataConnectionField: React.FC<ExistingDataConnectionFieldProps> =
...
+    connections.forEach((connection) => {
+      uniqueConnections.add(connection);
+    });
+
+    return Array.from(uniqueConnections).map((connection) => ({
+      value: connection.data.metadata.name,
+      content: getDataConnectionDisplayName(connection),
+      selected: connection.data.metadata.name === selectedDataConnection,
+    }));
+  }, [loaded, connections, selectedDataConnection]);
```
Comment (christianvogt) [inline frontend/src/pages/projects/screens/spawner/dataConnection/ExistingDataConnectionField.tsx:38]: Why was this change in logic necessary?

File: frontend/src/pages/projects/screens/spawner/storage/AddExistingStorageField.tsx [hunk 5]
```diff
@@ -23,23 +26,65 @@ const AddExistingStorageField: React.FC<AddExistingStorageFieldProps> = ({
...
+    storages.forEach((pvc) => {
+      uniqueIdentifiers.add(pvc);
+    });
+
+    return Array.from(uniqueIdentifiers).map((pvc) => ({
+      value: pvc.metadata.name,
+      content: getDisplayNameFromK8sResource(pvc),
+      selected: pvc.metadata.name === selectedStorage,
+    }));
+  }, [loaded, storages, selectedStorage]);
```
Comment (christianvogt) [inline frontend/src/pages/projects/screens/spawner/storage/AddExistingStorageField.tsx:51]: Same question: Why was this change in logic necessary?

Comment (jeff-phillips-18) [reply frontend/src/pages/projects/screens/spawner/storage/AddExistingStorageField.tsx]: fixed

File: frontend/src/pages/projects/screens/spawner/dataConnection/ExistingDataConnectionField.tsx (hunk 4 above)
Comment (jeff-phillips-18) [reply frontend/src/pages/projects/screens/spawner/dataConnection/ExistingDataConnectionField.tsx]: fixed

File: frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/ServingRuntimeSizeSection.tsx (hunk 3 above)
Comment (jeff-phillips-18) [reply frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/ServingRuntimeSizeSection.tsx:89]: fixed

File: frontend/src/pages/projects/screens/spawner/storage/AddExistingStorageField.tsx [hunk 6]
```diff
@@ -23,23 +26,65 @@ const AddExistingStorageField: React.FC<AddExistingStorageFieldProps> = ({
...
-      loaded={loaded}
-      loadError={loadError}
-      selectedStorage={data.storage}
-      setStorage={(storage) => setData({ ...data, storage: storage || '' })}
-      selectDirection={selectDirection}
-      menuAppendTo={menuAppendTo}
-    />
+      data-testid="persistent-storage-group"
+    >
+      <TypeaheadSelect
```
[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply frontend/src/pages/projects/screens/spawner/storage/AddExistingStorageField.tsx:59]: Fixed by updating the react-templates package

File: frontend/src/concepts/pipelines/content/tables/ExperimentSearchInput.tsx
[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply frontend/src/concepts/pipelines/content/tables/ExperimentSearchInput.tsx:1]: Thanks @emilys314 , I have updated the PF component and pulled it in. This should be fixed now.

File: frontend/src/pages/projects/notebook/ConnectedNotebookField.tsx
[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply frontend/src/pages/projects/notebook/ConnectedNotebookField.tsx:1]: Thanks @emilys314. Updated.

File: frontend/src/pages/modelRegistry/screens/ModelRegistrySelector.tsx [hunk 7]
```diff
@@ -82,46 +78,87 @@ const ModelRegistrySelector: React.FC<ModelRegistrySelectorProps> = ({
...
-          id={mr.metadata.name}
-          key={mr.metadata.name}
-          value={getResourceNameFromK8sResource(mr)}
-          description={getMRSelectDescription(mr)}
-          isFavorite={favorites.includes(mr.metadata.name)}
-        >
-          {getDisplayNameFromK8sResource(mr)}
-        </SelectOption>
-      ))}
+      <SelectList>
```
Comment (jenny-s51) [inline frontend/src/pages/modelRegistry/screens/ModelRegistrySelector.tsx:81]: This select list extends past with viewport width:

Screenshot 2024-08-02 at 11 13 27 AM

**Review by jenny-s51 — COMMENTED** [review]
These updates look great @jeff-phillips-18 , tested projects, pipelines, experiments, model registry and settings screens (couldn't test model serving either). 

Some fields use a different`Select` component, [SimpleDropdownSelect](https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/components/SimpleDropdownSelect.tsx#L24), alongside the new ones but this component doesn't apply any checkmark indicator to the selected items, is that expected? (see Notebook Image fields in "Create Workbench" form, for example)

We use that Select component in a lot of places throughout the app/forms - would it be worth opening a ticket to align the Select functionality throughout the dashboard? cc @christianvogt 

That would be out of scope of this work though, left a small comment below and everything else LGTM

**Comment by jeff-phillips-18** [comment]
Closing this PR in lieu of smaller PRs attacking the different usages of the select components (simple vs typeahead).

**Pull request closed without merging**